triage/2021-02-24.md

```diff
@@ -0,0 +1,43 @@
+# 2021-02-24 Triage Log
+
+Overall, a positive week for compiler performance with only one moderate regression. The change that introduced the regression leads to significantly improved [bootstrap speed](https://github.com/rust-lang/rust/pull/70951#issuecomment-766292996) of the compiler as well as easier maintainability.
+
+Triage done by **@rylev**.
+Revision range: [f1c47c79fe8438ed241630f885797eebef3a6cab..301ad8a4fa3ea56fb980443b7997c8f9d72dd717](https://perf.rust-lang.org/?start=f1c47c79fe8438ed241630f885797eebef3a6cab&end=301ad8a4fa3ea56fb980443b7997c8f9d72dd717&absolute=false&stat=instructions%3Au)
+
+1 Regression, 5 Improvements, 0 Mixed
+0 of them in rollups
+
+#### Regressions
+
+Move the query engine out of rustc_middle [#70951](https://github.com/rust-lang/rust/issues/70951)
+- Moderate regression in [instruction counts](https://perf.rust-lang.org/compare.html?start=e7c23ab933ebc1f205c3b59f4ebc85d40f67d404&end=83b30a639d5abd1270ade35d9bd92271f5a5ba18&stat=instructions:u) (up to 4.9% on `full` builds of `deeply-nested-check`)
+- While this does somewhat hurt compiler performance, it is a huge gain in bootstrap speed. The performance impact was deemed acceptable, but perhaps an investigation in if the remaining performance regressions can be eliminated, should be looked into.
+- Added a nag to [the pull request](https://github.com/rust-lang/rust/pull/70951#issuecomment-785044935).
+
+#### Improvements
+
+Only store a LocalDefId in some HIR nodes [#81611](https://github.com/rust-lang/rust/issues/81611)
+- Very large improvement in [instruction counts](https://perf.rust-lang.org/compare.html?start=a143517d44cac50b20cbd3a0b579addab40dd399&end=8fe989dd768f5dfdb0fc90933f3f74fa4579fefd&stat=instructions:u) (up to -9.6% on `incr-unchanged` builds of `many-assoc-items-check`)
+- Large improvements in the associated_item query which naturally led to improvements in the stress test related to associated items.
+
+Inline try_get_cached [#82197](https://github.com/rust-lang/rust/issues/82197)
+- Very large improvement in [instruction counts](https://perf.rust-lang.org/compare.html?start=8fe989dd768f5dfdb0fc90933f3f74fa4579fefd&end=ee88f46bb5e27c4d9f30326e69ff2298dcbeecb1&stat=instructions:u) (up to -10.4% on `full` builds of `externs-debug`)
+- An attempt to fix an issue due to a bad interaction with [#81892](https://github.com/rust-lang/rust/issues/81892) which removed a bunch of largely unnecessary inline directives.
```

Comment:
Should the formulation attribute the improvement to the three PRs ? https://github.com/rust-lang/rust/issues/82197 , https://github.com/rust-lang/rust/issues/81892 and https://github.com/rust-lang/rust/issues/81855

Reply:
The improvement itself is just from what was measured after the merging of rust-lang/rust#82197, but I can reference https://github.com/rust-lang/rust/pull/81855 if you'd like.

Reply:
The changes from rust-lang/rust#81855 are directly responsible for the improvement. I would appreciate if you could emphasize that. I would feel uncomfortable if the improvements were to be attributed to rust-lang/rust#82197 which merely added an inline hint to fix unintended interaction with rust-lang/rust#81892, which landed in the meantime. Thanks.

Reply:
@cjgillot @tmiasko I've adjusted the wording. Thoughts?

Reply:
I would describe it as:

> rust-lang/rust#82197 shows improvements originally expected from rust-lang/rust#81855, which based query fast path on `try_get_cached`. In-between the initial successful perf run of rust-lang/rust#81855 and merging, another PR rust-lang/rust#81892 removed an inline hint from the function, which was soon to be on the fast path.